llvm/lib/Transforms/Scalar/LoopDistribute.cpp

```diff
@@ -231,17 +231,16 @@ class InstPartition {
     }
   }
 
-  void print() const {
-    if (DepCycle)
-      dbgs() << "  (cycle)\n";
+  void print(raw_ostream &OS) const {
+    OS << (DepCycle ? " (cycle)\n" : "\n");
     for (auto *I : Set)
       // Prefix with the block name.
-      dbgs() << "  " << I->getParent()->getName() << ":" << *I << "\n";
+      OS << "  " << I->getParent()->getName() << ":" << *I << "\n";
   }
 
-  void printBlocks() const {
+  void printBlocks(raw_ostream &OS) const {
```

Comment:
Is adding the output stream as arg needed? AFAICT it will only be `dbgs()` with this patch?

Reply:
The `print()` function on line 530 takes an output stream, so I thought I should change the other `print` function and `printBlocks` to take an output stream as well, for uniformity. Besides, this is also the convention used in other `dump()` functions in the tree.

Reply:
I additionally investigated stripping the `raw_ostream` argument everywhere, but this is not possible due to `operator<<`.